### PR TITLE
ch_tests_tool: Improve subtests result parsing logic

### PR DESCRIPTION
**Pull request by pupacha**
At times, we see multiple jsons present in a single line output because of which subtest result is not being parsed into json and is ignored. This will change will take care of extracting multiple JSON objects from a single line of string and none of the subtest results extraction is missed.
Ex line: {"version":"30.0.0"}{ "type": "test", "name": "common_parallel::test_boot_from_virtio_pmem", "event": "ok" }

### CODE REVIEW

**Review by squirrelsc — DISMISSED** [review]

File: microsoft/testsuites/cloud_hypervisor/ch_tests_tool.py [hunk 1]
```diff
@@ -250,49 +250,39 @@ def _extract_test_results(
         # { "type": "test", "event": "ok", "name": "integration::test_vfio" }
         lines = output.split("\n")
         for line in lines:
-            matches = re.findall(r"{.*}", line)
-            if not matches:
+            if '"type": "test"' not in line:
```
Comment (anirudhrb) [inline microsoft/testsuites/cloud_hypervisor/ch_tests_tool.py]: Why do we need this? This is not a reliable check. In the future, cargo might format the JSON without the space i.e. `"type":"test"`.

File: microsoft/testsuites/cloud_hypervisor/ch_tests_tool.py [hunk 2]
```diff
@@ -250,49 +250,39 @@ def _extract_test_results(
...
-
-            if result["event"] == "started":
-                status = TestStatus.RUNNING
-            elif result["event"] == "ok":
-                status = TestStatus.PASSED
-            elif result["event"] == "failed":
-                status = TestStatus.FAILED
-            elif result["event"] == "ignored":
-                status = TestStatus.SKIPPED
+            for result in extract_jsons(line):
```
Comment (anirudhrb) [inline microsoft/testsuites/cloud_hypervisor/ch_tests_tool.py]: We discussed that we would first try to parse the line as-is and if that fails, we will try `extract_jsons`. Why did you deviate from that?

File: microsoft/testsuites/cloud_hypervisor/ch_tests_tool.py (hunk 1 above)
Comment (pupacha) [reply microsoft/testsuites/cloud_hypervisor/ch_tests_tool.py]: ok, I'll remove this check. This check isn't a necessity.

File: microsoft/testsuites/cloud_hypervisor/ch_tests_tool.py (hunk 2 above)
Comment (pupacha) [reply microsoft/testsuites/cloud_hypervisor/ch_tests_tool.py]: Changed it to parse the line as-is initially and try `extract_jsons` for failing case.

**Review by anirudhrb — APPROVED** [review]

**Comment by pupacha** [comment]
@squirrelsc @LiliDeng Please review

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**